src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityReplyChannelBinder.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                context = HandleHandshake(context);
+                //Add logic to separate flow for handshake or actual call
+                return _next.DispatchAsync(context);
+            }
+            else
+            {
+                return ProcessRequest(context);
+
+            }
+
+        }
+
+        public Task DispatchAsync(Message message)
+        {
+            throw new NotImplementedException();
+        }
+
+
+
+
+
+        private Task ProcessRequest(RequestContext context)
+        {
+            String sessionKey = GetIdentifier(context.RequestMessage.Headers);
+            if (String.IsNullOrEmpty(sessionKey))
+                throw new Exception("Session Key is mising");
+            if (this._securityDispatcher.SessionServerSettings != null)
+            {
+                UniqueId sessionId = new UniqueId(sessionKey);
+                SecuritySessionServerSettings settings = this._securityDispatcher.SessionServerSettings;
+                SecurityContextSecurityToken cacheToken =
+                    settings.GetSecurityContextSecurityToken(sessionId);
+                SecurityContextSecurityToken securityToken = cacheToken.Clone();
+                settings.RemovePendingSession(sessionId);
+                ServerSecuritySimplexSessionChannel.SecurityReplySessionChannel
+                    sessionReplyChannel = new ServerSecuritySimplexSessionChannel.
+                    SecurityReplySessionChannel(settings, securityToken, null
+                    , settings.SettingsLifetimeManager);
+                RequestContext securityRequestContext = sessionReplyChannel.ReceiveRequest(context);
+                ServiceDispatcher serviceDispatcher = (ServiceDispatcher)this._securityDispatcher.InnerServiceDispatcher;
+                Task<IServiceChannelDispatcher> serviceChannelDispatcherTask = serviceDispatcher.CreateServiceChannelDispatcherAsync(this._securityDispatcher.OuterChannel);
+                IServiceChannelDispatcher serviceChannelDispatcher = serviceChannelDispatcherTask.GetAwaiter().GetResult();
+                return serviceChannelDispatcher.DispatchAsync(securityRequestContext);
+            }
+            else
+            {
+                throw new Exception("Session in server missing");
+            }
+
+        }
+
+        private SecurityRequestContext HandleHandshake(RequestContext context)
+        {
+            SecurityReplyChannel securityReplyChannel = (SecurityReplyChannel)_channel;
+            SecurityRequestContext securedMessage = (SecurityRequestContext)securityReplyChannel.ProcessReceivedRequest(context);
+            return securedMessage;
+        }
+
+
+        private String GetIdentifier(MessageHeaders headers)
+        {
+            int headerIndex = headers.FindHeader(XD.SecurityJan2004Dictionary.Security.Value, XD.SecurityJan2004Dictionary.Namespace.Value);
+            if(headerIndex >0)
```

Comment:
I'm concerned about this line. I'm not finding a reference to this in the original code. It looks like this should be referencing SecurityVersion.HeaderName and SecurityVersion.HeaderNamespace. This is hard coded for a specific SecurityVersion and this needs to be more generic. The same thing applies to the code around SecureConversation as there are multiple SecureConversation versions.